File: leetcode/Python/backspace-string-compare.py
class Solution:
    def backspaceCompare(self, s: str, t: str) -> bool:
        def transString(st):
            sol = list()
            for ch in st:
                if ch == '#' and len(sol) > 0:
                    sol.pop()
                elif ch != '#':
                    sol.append(ch)
            return ''.join(sol)
        return transString(s) == transString(t)
                    

File: leetcode/C++/backspace-string-compare.cpp
class Solution {
public:
    string transString(string &s){
        string st;
        for(char &ch : s){
            if(ch=='#' && !st.empty()) st.pop_back();
            else if(ch != '#') st.push_back(ch);
        }
        return st;
    }
    bool backspaceCompare(string s, string t) {
        return transString(s) == transString(t);
    }
};